crates/hotshot-state-prover/src/circuit.rs

```diff
@@ -0,0 +1,219 @@
+//! Circuit implementation for verifying light client state update
+
+use std::marker::PhantomData;
+
+use ark_ec::twisted_edwards::TECurveConfig;
+use ark_ff::PrimeField;
+use ethereum_types::U256;
+use hotshot_stake_table::config::STAKE_TABLE_CAPACITY;
+use hotshot_types::traits::{
+    stake_table::{SnapshotVersion, StakeTableScheme},
+    state::HotShotStateForProver as HotShotState,
+};
+use jf_plonk::errors::PlonkError;
+use jf_primitives::{
+    circuit::signature::schnorr::VerKeyVar,
+    rescue::RescueParameter,
+    signatures::{
+        bls_over_bn254::VerKey as BLSVerKey,
+        schnorr::{Signature, VerKey as SchnorrVerKey},
+    },
+};
+use jf_relation::{
+    errors::CircuitError, gadgets::ecc::TEPoint, BoolVar, Circuit, PlonkCircuit, Variable,
+};
+
+/// convert a U256 to a field element.
+pub(crate) fn u256_to_field<F: PrimeField>(v: &U256) -> F {
```

Comment:
What if `v` is larger than the field size? Should we return an error in that case?

Reply:
This is only for stake amount, and we assume that it fits into the native field. Maybe I should call it "lossy conversion" for clarity.